src/mixins/object_geometry.mixin.js

```diff
@@ -183,67 +202,112 @@
 
     /**
      * Checks if object is partially contained within the canvas with current viewportTransform
      * @param {Boolean} [calculate] use coordinates of current position instead of .oCoords
      * @return {Boolean} true if object is partially contained within canvas
      */
-    isPartiallyOnScreen: function(calculate) {
+    isPartiallyOnScreen: function (calculate) {
       if (!this.canvas) {
         return false;
       }
-      var pointTL = this.canvas.vptCoords.tl, pointBR = this.canvas.vptCoords.br;
-      if (this.intersectsWithRect(pointTL, pointBR, true, calculate)) {
-        return true;
+      var tl = this.canvas.vptCoords.tl, br = this.canvas.vptCoords.br;
+      return this.isPartiallyOnBoundingBox(tl, br, true, calculate);
+    },
+
+    /**
+     * Checks if object's shadow is displayed within the canvas with current viewportTransform
+     * the check is done stopping at first point that appears on screen
+     * @param {Boolean} [calculate] use coordinates of current position instead of .aCoords
+     * @return {Boolean} true if object is fully or partially contained within canvas
+     */
+    isShadowOnScreen: function (calculate) {
+      if (!this.canvas || !this.shadow) {
+        return false;
       }
-      var allPointsAreOutside = this.getCoords(true, calculate).every(function(point) {
-        return (point.x >= pointBR.x || point.x <= pointTL.x) &&
-        (point.y >= pointBR.y || point.y <= pointTL.y);
-      });
-      return allPointsAreOutside && this._containsCenterOfCanvas(pointTL, pointBR, calculate);
+      var shadowOffsets = this.getShadowOffsets(),
+          shadowOffset = shadowOffsets.offset, blurOffset = shadowOffsets.blur;
+      //  instead of offsetting object coordinates to calculate shadows coordinates,
+      //  we translate canvas' corners to the shadow's coordinate system
+      var tl = this.canvas.vptCoords.tl.subtract(shadowOffset).subtract(blurOffset),
+          br = this.canvas.vptCoords.br.subtract(shadowOffset).add(blurOffset);
+      return this.isOnBoundingBox(tl, br, true, calculate);
+    },
+
+    /**
+     * Checks if object's shadow is partially contained within the canvas with current viewportTransform
+     * @param {Boolean} [calculate] use coordinates of current position instead of .oCoords
+     * @return {Boolean} true if object is partially contained within canvas
+     */
+    isShadowPartiallyOnScreen: function (calculate) {
+      if (!this.canvas || !this.shadow) {
+        return false;
+      }
+      var shadowOffsets = this.getShadowOffsets(),
+          shadowOffset = shadowOffsets.offset, blurOffset = shadowOffsets.blur;
+      //  instead of offsetting object coordinates to calculate shadows coordinates,
+      //  we translate canvas' corners to the shadow's coordinate system
+      var tl = this.canvas.vptCoords.tl.subtract(shadowOffset).subtract(blurOffset),
+          br = this.canvas.vptCoords.br.subtract(shadowOffset).add(blurOffset);
```

Comment:
should this be
`this.canvas.vptCoords.br.add(shadowOffset).add(blurOffset)` ?

Reply:
We translate vptCoords to the shadow coord system, so all points need to move by the opposite of the shadow offset.

Reply:
Hope this explains it well enough